### ISSUE TITLE
Structural Editing: move form left/right

### ISSUE THREAD

**Issue by pazustep** [cursive-ide/cursive#170]
It would be nice to have a structural editing action to move forms between their siblings. For example (with ⏐ representing the cursor):

Before: `(map ⏐coll fn) ; oops, it's the other way around` 
After: `(map fn ⏐coll) ; ah, much better`

This would only move adjacent forms in the same level, not barf, slurp or raise them.
